app/src/org/commcare/utils/FormUploadUtil.java

```diff
@@ -134,13 +128,12 @@ public static FormUploadResult sendInstance(int submissionNumber, File folder,
         // If we're listening, figure out how much (roughly) we have to send
         long bytes = estimateUploadBytes(files);
 
-        if (hasListener) {
-            myListener.startSubmission(submissionNumber, bytes);
+        if (listener != null) {
+            listener.startSubmission(submissionNumber, bytes);
         }
 
         if (files.length == 0) {
             Log.e(TAG, "no files to upload");
-            listener.cancel(true);
```

Comment:
Removed this since we should not cancel upload for all forms because of an error with a single form and allow code to follow the [error handling here](https://github.com/dimagi/commcare-android/blob/formSubmissionsScheduling/app/src/org/commcare/sync/FormSubmissionHelper.java#L188-L202) for this case.

Reply:
> Specifically: There are lots of situations where we should cancel an upload because of an error with a single form. If there's a form we're having issues with that we don't understand, if we block submission we retain the synchronicity between the mobile and the server until we can correct it. If we let stuff through we risk letting the phone and server drift significantly.

@ctsims Agree that we might want to cancel the whole task in some specific error in a single form. Though talking specifically for this case, the [error handling](https://github.com/dimagi/commcare-android/blob/formSubmissionsScheduling/app/src/org/commcare/sync/FormSubmissionHelper.java#L316)  (I put a wrong link earlier) seems more robust as it takes into account whether the FNF is due to storage not being available or not and halt/continue the form upload for all forms depending on that which seems more robust to me than cancelling the task in all cases. Would we still want to cancel the upload for all forms even if the issue is with one form for the sake of mobile server parity ?

Reply:
@ctsims  bump

Reply:
@shubham1g5 You are right that it looks like the general form processing loop will catch a FileNotfound error correctly if storage isn't available, so agreed that we can let it propagate and not catch it redundantly..